crates/driver/src/domain/competition/auction.rs

```diff
@@ -41,6 +42,7 @@ impl Auction {
         tokens: impl Iterator<Item = Token>,
         deadline: time::Deadline,
         eth: &Ethereum,
+        surplus_capturing_jit_order_owners: &[eth::Address],
```

Comment:
This can already take a `HashSet<eth::Address>` to avoid cases where you currently collect values into a vector, pass a reference and then collect into a hashset.